### PR TITLE
Neka manager form style, functionality

### LINKED ISSUE #99: Neka - Save Listing upon Manager Submission of Listing Form
Currently, our platform lacks the functionality to save a listing when a manager submits a listing form. To improve our system, we need to implement a feature that allows managers to submit listing information, which will then be saved in our database.

Acceptance Criteria:

- Create a function to handle the submission of a listing form by managers.
- Implement validation logic to ensure that all required fields in the listing form are provided before saving the listing.
- Upon successful submission, save the listing information in the Listings collection of our database.

### PR DESCRIPTION
**Pull request by istoga**
## What type of PR is this? (check all applicable)

- [ ] Feature
- [ ] Refactor
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [X] Other (please specify here: style manager form
  
  
## Description of what this Pull Request does
style rest of manager form
will still need to add tailwind css

## Related Tickets & Documents
- Closes #99 #120 
- Related Issues: #120
  
## QA Instructions, Screenshots, Recordings
Tailwind, still being added

https://github.com/AritDeveloperCircle/findyourpeople/assets/57186734/c781b6e2-2269-4b84-bb91-8a1b36d2d814

  
## Added/updated tests?
- [ ] Yes
- [ ] Not Yet, and here is why: __explain here__
- [ ] I need help with tests
- [X] No, and here is why:no testing

### CODE REVIEW

**Comment by TaylorDJones11** [comment]
Looks good! @istoga 
My only suggestion would be about the vision/description fields instead of inputs making them textarea. The input looks like the user may be limited to characters, whereas a textarea shows the user there is more space allocated.
Screenshot 2023-06-08 at 11 30 32

**Comment by istoga** [comment]
@TaylorDJones11 that's a great point, I will change that

File: src/pages/managerForm.js
Comment (msarit) [inline src/pages/managerForm.js:132]: Do we have the field in the database for this "Choose Category" question? If not, please omit for the MVP. 🙏🏾 
If we do have this field in the DB, are we going to allow for managers to create their own categories if none of t suggested categories fit? If not, then we should have an "Other" option.

I'm fine with either option; I just have the end-of-June in sight 😅

File: src/pages/managerForm.js
Comment (msarit) [inline src/pages/managerForm.js:187]: Let this read "Community Manager Name"

File: src/pages/managerForm.js
Comment (msarit) [inline src/pages/managerForm.js:198]: Let this read "Year Established"

File: src/pages/managerForm.js
Comment (msarit) [inline src/pages/managerForm.js:211]: Do we have a "Location" field in the DB?

If not, remove from form.

If yes, I think we should add a description under the field that reads: _If you host in-person events, type your City/State and/or Country. If not, type 'Online'_

File: src/pages/managerForm.js
Comment (msarit) [inline src/pages/managerForm.js:237]: Actually the linkedin URL is for the Community Manger, not the Community.
Move this field to right after the Community Manager Name field.

File: src/pages/managerForm.js
Comment (msarit) [inline src/pages/managerForm.js:249]: Actually the Twitter URL is for the Community Manger, not the Community.
Move this field to right after the Community Manager LinkedIn field.

File: src/pages/managerForm.js
Comment (msarit) [inline src/pages/managerForm.js:261]: Remove this field; we are not collecting Facebook info

**Review by msarit — CHANGES_REQUESTED** [review]

File: src/pages/managerForm.js
Comment (istoga) [reply src/pages/managerForm.js:132]: I see your point. Removed.

File: src/pages/managerForm.js
Comment (istoga) [reply src/pages/managerForm.js:187]: Done

File: src/pages/managerForm.js
Comment (istoga) [reply src/pages/managerForm.js:198]: Done

File: src/pages/managerForm.js
Comment (istoga) [reply src/pages/managerForm.js:211]: We don't, I will remove.

File: src/pages/managerForm.js
Comment (istoga) [reply src/pages/managerForm.js:261]: Done

File: src/pages/managerForm.js
Comment (jjleigh) [inline src/pages/managerForm.js:1]: This file is getting long. Is it possible to break it into smaller components based on sections?

File: src/pages/managerForm.js
Comment (jjleigh) [inline src/pages/managerForm.js:84]: I think you can create a helper file where all these file-related functions can live and then just be imported here. This will make it easier for them to be tested and reuseable.

**Review by jjleigh — CHANGES_REQUESTED** [review]

**Comment by Domicaren1** [comment]
@istoga have you started working on the styling? If no, I can take it up.

**Comment by istoga** [comment]
@Domicaren1  I started, but please feel free to continue, we want to get everything finished on time. Thanks for offering!

**Comment by istoga** [comment]
Thanks @ibimina for your help, greatly appreciated.

File: src/pages/managerForm.js
Comment (istoga) [reply src/pages/managerForm.js]: Done.

File: src/pages/managerForm.js
Comment (istoga) [reply src/pages/managerForm.js]: Done

**Comment by Domicaren1** [comment]
> 

Ok @istoga . I'll start. Thanks

**Comment by msarit** [comment]
@Domicaren1 @istoga Please let me know when this PR is ready for review 🙏🏾

File: src/pages/managerForm.js
Comment (istoga) [reply src/pages/managerForm.js]: Created function to loop over inputs with .map

File: src/pages/managerForm.js
Comment (istoga) [reply src/pages/managerForm.js]: moved helper functions into own file and imported to use in managerForm

**Comment by istoga** [comment]
@msarit I finished the functionality and cleaned up the code for creation, but in order to do editing, it looks like that should happen over on the manager dashboard with a button. I will head over there and place a button on the dashboard. And will work on the editing portion. If there any suggestions, or I am not heading in the right way, please, anyone let me know, thanks!

File: src/pages/helperFile.js
Comment (jjleigh) [inline src/pages/helperFile.js:1]: Good job :). I would create a folder called helpers and then rename this file to be FileHelper. Just so it can have a single responsibility for file-related functions.

File: src/pages/managerForm.js
Comment (jjleigh) [inline src/pages/managerForm.js:116]: Well done 🥇

File: src/pages/managerForm.js
Comment (jjleigh) [inline src/pages/managerForm.js:37]: I think this can be extracted to a ManagerFormHelper file. Create a function that returns the appropriate fields. This will make it easier to test. Then you can put other manager form related functions.

File: src/pages/managerForm.js
Comment (jjleigh) [inline src/pages/managerForm.js:17]: I think this could also be extracted to a ManagerFormHelper file. Not super important but if you are going to create one based on the other comment below then this could go in there too.

File: src/pages/managerForm.js
Comment (jjleigh) [inline src/pages/managerForm.js:137]: I think this could be a separate component that can be imported here. ImageUploader component.

Comment (istoga) [reply src/pages/managerForm.js]: Removed due to descoping.

File: src/pages/helperFile.js
Comment (istoga) [reply src/pages/helperFile.js:1]: Since we are descoping the file upload, it looks like we really don't need a helper file anymore. After removing the functions associated with file upload, I only had one left, and that can be taken care of with an anonymous function on the button itself.
``<button
            className={styles.returnButton}
            onClick={() => router.push("/")}
            text="Return to Home"
            type="submit"
          >``

File: src/pages/managerForm.js
Comment (istoga) [reply src/pages/managerForm.js:116]: Thank you so much!

File: src/pages/managerForm.js
Comment (istoga) [reply src/pages/managerForm.js:17]: Due to time constraints, I will defer this until all other issues are done.

**Review by jjleigh — APPROVED** [review]

**Review by msarit — COMMENTED** [review]
I left some comments for changes that I believe should be made, but because we are down to the wire, I won't Request Changes, thereby blocking the merge.

@istoga If you would like me to merge without my suggested changes please let me know.

Also I noticed that:
(1) Creating a listing does not produce any confirmation message, that lets me know the listing was created successfully, and that it must be approved before it appears on the front page.
(2) Does the form store the created listing in the database? I checked and did not see a new listing there.

Again I will merge if we want the work committed as-is. Let me know! 🙏🏾

File: src/pages/managerForm.js
Comment (istoga) [reply src/pages/managerForm.js:48]: Done

File: src/pages/managerForm.js
Comment (istoga) [reply src/pages/managerForm.js:51]: Done

File: src/pages/managerForm.js
Comment (istoga) [reply src/pages/managerForm.js:72]: Done

**Comment by istoga** [comment]
I added a simple alert box so users would know that their community was added to the database. I also checked and it looks like the one I put in actually did update to the database. I entered one called Yellowstone National Park.

**Review by msarit — APPROVED** [review]

**Pull request merged**
